packages/venia-concept/src/components/SearchBar/autocomplete.js

```diff
@@ -1,154 +1,79 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import debounce from 'lodash.debounce';
-import { Query } from 'src/drivers';
-
-import classify from 'src/classify';
-import { loadingIndicator } from 'src/components/LoadingIndicator';
-import SuggestedCategories from './suggestedCategories';
-import SuggestedProducts from './suggestedProducts';
-import PRODUCT_SEARCH from '../../queries/productSearch.graphql';
-
+import React, { useEffect } from 'react';
+import { bool, func, shape, string } from 'prop-types';
+import { useFieldState } from 'informed';
+import { useQuery } from '@magento/peregrine';
+
+import { mergeClasses } from 'src/classify';
+import PRODUCT_SEARCH from 'src/queries/productSearch.graphql';
+import Suggestions from './suggestions';
 import defaultClasses from './autocomplete.css';
 
-const debounceTimeout = 200;
-const suggestedCategoriesLimit = 4;
-const suggestedProductsLimit = 3;
+const DEBOUNCE_WAIT = 200;
 
-class SearchAutocomplete extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string,
-            statusContent: PropTypes.string
-        }),
-        searchQuery: PropTypes.string.isRequired,
-        autocompleteVisible: PropTypes.bool,
-        executeSearch: PropTypes.func.isRequired,
-        updateAutocompleteVisible: PropTypes.func.isRequired
-    };
+const Autocomplete = props => {
+    const { setVisible, visible } = props;
 
-    constructor(props) {
-        super(props);
-        this.state = {
-            isQueryUpdating: false,
-            autocompleteQuery: props.searchQuery || ''
-        };
-    }
+    const [queryResult, queryApi] = useQuery(PRODUCT_SEARCH, DEBOUNCE_WAIT);
+    const { data, error, loading } = queryResult;
+    const { resetState, runQuery, setLoading } = queryApi;
 
-    componentDidUpdate = prevProps => {
-        const { searchQuery } = this.props;
+    const { value } = useFieldState('search_query');
+    const valid = value && value.length > 2;
```

Comment:
Of what significance is a length of 2?

Reply:
Not sure; this is preserved from the original implementation, and seems reasonable. `3` represents the minimum length a search term must have before the query will be run.